pinot-common/src/main/java/com/linkedin/pinot/common/segment/crypt/MultipartPinotCrypter.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.segment.crypt;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.IOUtils;
+import org.glassfish.jersey.media.multipart.FormDataBodyPart;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The MultiPartPinotCrypter takes in a multipart object and extracts the tarred segment file.
+ */
+public class MultipartPinotCrypter implements PinotCrypter {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MultipartPinotCrypter.class);
+
+  @Override
+  public void encrypt(Object decryptedObject, File encryptedFile) throws Exception {
+    throw new UnsupportedOperationException("Multipart Pinot Crypter does not support encryption");
```

Comment:
I don't think we need it because we would never want to transform a pinot segment to a multipart in our code I think but because we have a crypter (we want the interface to support decrypt and encrypt), I left the encrypt method there.